private-templates-service/client/src/store/currentTemplate/types.ts

```diff
@@ -0,0 +1,34 @@
+import { IncomingMessage } from "http";
+
+export interface CurrentTemplateState {
+  templateID: string | undefined;
+  templateJSON: string | undefined;
+  templateName: string | undefined;
+  sampleDataJSON: string | undefined;
+  isFetching: boolean;
+}
+
+//Action Types
+export const NEW_TEMPLATE = 'NEW_TEMPLATE';
+export const REQUEST_NEW_TEMPLATE_UPDATE = 'REQUEST_NEW_TEMPLATE_UPDATE';
+export const RECEIVE_NEW_TEMPLATE_UPDATE = 'RECEIVE_NEW_TEMPLATE_UPDATE';
+export const FAILURE_NEW_TEMPLATE_UPDATE = 'FAILURE_NEW_TEMPLATE_UPDATE';
+export const REQUEST_EXISTING_TEMPLATE_UPDATE = 'REQUEST_EXISTING_TEMPLATE_UPDATE';
+export const RECEIVE_EXISTING_TEMPLATE_UPDATE = 'RECEIVE_EXISTING_TEMPLATE_UPDATE';
+export const FAILURE_EXISTING_TEMPLATE_UPDATE = 'FAILURE_EXISTING_TEMPLATE_UPDATE';
+
+export interface CurrentTemplateAction {
+  type: typeof NEW_TEMPLATE |
```

Comment:
rename to 'actionType' and 'actionText'

Reply:
haha so I changed it to actionType but then realized that redux requires this to be type